webapi/src/test/scala/org/knora/webapi/responders/admin/ProjectsServiceLiveSpec.scala

```diff
@@ -1,91 +1,177 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
 package org.knora.webapi.responders.admin
-import zio.Scope
-import zio.ZIO
+
 import zio._
 import zio.mock._
-import zio.test.Assertion
-import zio.test.Spec
-import zio.test.TestEnvironment
-import zio.test.ZIOSpecDefault
-import zio.test.assertTrue
+import zio.prelude.Validation
+import zio.test._
+
+import java.util.UUID
 
+import dsp.util.UuidGenerator
+import dsp.valueobjects.Iri._
 import dsp.valueobjects.Project.ShortCode
-import org.knora.webapi.messages.ResponderRequest
+import dsp.valueobjects.Project._
+import dsp.valueobjects.V2
 import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectCreatePayloadADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectCreateRequestADM
 import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectGetRequestADM
 import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectGetResponseADM
-import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectIdentifierADM.ShortcodeIdentifier
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectIdentifierADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectOperationResponseADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectsGetRequestADM
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectsGetResponseADM
 import org.knora.webapi.messages.store.triplestoremessages.StringLiteralV2
+import org.knora.webapi.messages.util.KnoraSystemInstances
 import org.knora.webapi.responders.ActorToZioBridge
 import org.knora.webapi.responders.ActorToZioBridgeMock
 import org.knora.webapi.responders.admin.ProjectsService
 
 object ProjectsServiceLiveSpec extends ZIOSpecDefault {
 
   private val expectNoInteraction = ActorToZioBridgeMock.empty
-  // private val systemUnderTest     = ZIO.service[ProjectsService]
 
-  final case class BridgeMock() extends ActorToZioBridge {
-    var msg: ResponderRequest = null
-
-    override def askAppActor[R: Tag](message: ResponderRequest): Task[R] = {
-      msg = message
-      println("msg", msg)
-      println("type", msg.getClass())
-      ZIO.attempt(null.asInstanceOf[R])
-    }
+  val uuidGenerator = new UuidGenerator {
+    override def createRandomUuid: UIO[UUID] = for {
+      random <- ZIO.random
+      _      <- random.setSeed(0) // makes the random deterministic: bb20b45f-d4d9-4138-bd93-cb799b3970be
+      uuid   <- random.nextUUID
+    } yield uuid
```

Comment:
See my comment above as to why I think the `UuidGenerator` should be deprecated with this approach of managing randomness.